tests/test_visitors/test_ast/test_general/test_comparison_order.py

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+
+import pytest
+
+from wemake_python_styleguide.violations.consistency import (
+    ComparisonOrderViolation,
+)
+from wemake_python_styleguide.visitors.ast.order import WrongOrderVisitor
+
+@pytest.mark.parametrize('code', [
```

Comment:
After thinking for a while about the testing strategy, I have came with the solution:

0. There are different cases and types that should be checked
1. We should create template for different cases at the module level, like here: https://github.com/wemake-services/wemake-python-styleguide/blob/master/tests/test_visitors/test_ast/test_general/test_magic_numbers.py#L13
2. We should format these templates with different values inside the test

Templates to be checked:
- `if {0} is {1}: ...`, and for `is not`
- `if {0} == {1}: ...`, and for `!=`
- `if {0} > {1}: ...`, and for `<`, `<=`, `>=`
- `ternary = 5 if {0} > {1} else 6`, we only need to check that ternary operators work
- `while {0} > {1}:`, we need to check that `while` also works

Values to be inserted:
- variables (should always work)
- integers, floats, booleans (should work when located on the right side)
- strings (should work when located on the right side)
- tuples, lists, dicts, sets

And then we have special cases:
- `if {0} > {1} and {0} < {1}:` this should raise two errors
- `if {0} < {1} < {2}:` here we should insert different cases of comparison. Like `0, x, 4`and `x, 0, y` should work, while `0, 5, x` and `0, 5, 6` should not.